src/main/daml/Daml/Finance/Instrument/Swap/Fpml/Util.daml

```diff
@@ -83,13 +84,49 @@ adjustDateAccordingToBusinessDayAdjustments unadjustedDate businessDayAdjustment
     NoAdjustment -> pure $ unadjustedDate
     _ -> getCalendarsAndAdjust unadjustedDate businessDayAdjustments issuer calendarDataAgency
 
+-- | Create the claims for fixed rate payments.
+-- This function supports amortizing notionals.
+-- It will be integrated with the floating utility functions soon, since they have more and more code in common.
+createFixRateAmortizingNotionalPaymentClaimsList : SwapStream -> [SchedulePeriod] -> PeriodicSchedule -> Bool -> Decimal -> Bool -> DayCountConventionEnum -> Decimal -> Deliverable -> [Claim Date Decimal Deliverable Observable]
+createFixRateAmortizingNotionalPaymentClaimsList s calculationSchedule periodicSchedule useAdjustedDatesForDcf couponRate issuerPays dayCountConvention notional cashInstrumentCid = do
+  let
+    c = s.calculationPeriodAmount.calculation
+    n = fromSome c.notionalSchedule
+    notionalSteps = n.notionalStepSchedule.step
+    check = if length notionalSteps == length calculationSchedule then
+              if any (\(n, p) -> n.stepDate /= p.unadjustedStartDate) $ zip notionalSteps calculationSchedule then error("notional step schedule does not match calculationSchedule")
+              else True
+            else error("Number of notional steps do not match the number of calculationSchedule periods")
+  let
+    couponDatesAdjusted = map (.adjustedEndDate) calculationSchedule
+    couponAmounts = map (\p -> couponRate * (calcPeriodDcf dayCountConvention p useAdjustedDatesForDcf periodicSchedule.terminationDate periodicSchedule.frequency)) calculationSchedule
+    couponClaimAmounts = mconcat $ zipWith3 (\d a n -> when (TimeGte d) $ scale (Const a) $ scale (Const n.stepValue) $ one cashInstrumentCid) couponDatesAdjusted couponAmounts notionalSteps
+    couponClaims = if issuerPays then couponClaimAmounts else give couponClaimAmounts
+  pure couponClaims
+
+-- | Calculate a fix rate amount for each payment date and create claims.
+-- This function supports amortizing notionals.
+-- It will be integrated with the floating utility functions soon, since they have more and more code in common.
+createFixRateAmortizingNotionalPaymentClaims : SwapStream -> [SchedulePeriod] -> PeriodicSchedule -> Bool -> Decimal -> Bool -> DayCountConventionEnum -> Decimal -> Deliverable -> TaggedClaim
+createFixRateAmortizingNotionalPaymentClaims s schedule calculationSchedule useAdjustedDatesForDcf couponRate issuerPays dayCountConvention notional cashInstrumentCid = do
```

Comment:
No need for `do` block

Reply:
This function will be replaced soon.

Reply:
(As part of this PR: https://github.com/digital-asset/daml-finance/issues/286)